src/features/profile/mySecurity/backupCryptoKey/CryptoKey.tsx

```diff
@@ -1,45 +1,177 @@
+  },
+  mainImg: {
+    width: '100%',
+  },
+})
+
+const PDFDocument = ({
+  allPrivateKeys,
+  currencyName,
+  title,
+}: TPDFDocumentProps) => {
   return (
-    <Document title='Crypto Keys'>
-      <Page size='A4'>
-        <View>
-          <Text>Private Key: {publicKey}</Text>
-        </View>
-        <View>
-          <Text>Secret Key: {secretKey}</Text>
-        </View>
-      </Page>
+    <Document title={title}>
+      {allPrivateKeys?.map((privateKey, idx) => (
+        <Page size='A4' key={idx} style={pdfStyles.page}>
+          <View style={pdfStyles.section}>
+            <View style={pdfStyles.contentTop}>
+              <View style={pdfStyles.topMedia}>
+                <QRCodeGEnerator address={privateKey.address} />
+              </View>
+              <View style={pdfStyles.contentWrapper}>
+                <Text style={pdfStyles.contentTitle}>
+                  {currencyName} Address (Sapling)
+                </Text>
+                <Text style={pdfStyles.contentValue}>
+                  {addLineBreakForContent(privateKey.address)}
+                </Text>
+              </View>
+            </View>
+            <View style={pdfStyles.mainContent}>
+              <View style={pdfStyles.mainContentWrapper}>
+                <View style={pdfStyles.contentItem}>
+                  <Text style={pdfStyles.contentTitle}>Private Key</Text>
+                  <Text style={pdfStyles.contentValue}>
+                    {addLineBreakForContent(privateKey.privateKey)}
+                  </Text>
+                </View>
+                <View style={pdfStyles.marginTop20}>
+                  <Text style={pdfStyles.contentTitle}>
+                    {currencyName} Address (Sapling)
+                  </Text>
+                  <Text style={pdfStyles.contentValue}>
+                    {addLineBreakForContent(privateKey.address)}
+                  </Text>
+                </View>
+              </View>
+              <View style={pdfStyles.mainMedia}>
+                <QRCodeGEnerator address={privateKey.address} />
+              </View>
+            </View>
+          </View>
+        </Page>
+      ))}
     </Document>
   )
 }
 
 const CryptoKey = (props: TCrypto): JSX.Element => {
-  const { currencyName } = props
-  const publicKey =
-    'ps19jxlfdl8mhnsqlxlfdl8mhnsqlf7x0cwlhx0cwlhf7x0cwlhx0cwlheq0v34'
-  const secretKey =
-    'ps19jxlfdl8mhnsqlf7x0cwxlfdl8mhnsqlf7x0cwlhx0cwlhlhx0cwlheq0v34'
+  const { currencyName, rpcConfig } = props
+  const [allPrivateKeys, setAllPrivateKeys] = useState<TPrivateKey[]>([])
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const result = await fetcAllPrivateKeys(rpcConfig)
+      setAllPrivateKeys(result)
+    }
+    if (!allPrivateKeys.length) {
+      fetchData()
+    }
+  }, [])
+
+  const generateFileName = () => {
+    return `${currencyName || 'LSP'}_Paper_Wallet__Private_Keys_${dayjs(
+      new Date(),
+    ).format('MM_DD_YYYY__HH_MM_ss')}`
+  }
```

Comment:
NB: Better to have a variable `fileName` instead of function `generateFileName` so it won't be calculated twice per render